analysis/deathknight/src/RuneBreakdown.js

```diff
@@ -38,134 +39,126 @@ class RuneBreakdown extends ResourceBreakdown {
-                        style={{ width: `${(ability.generated / totalGenerated) * 100}%` }}
-                      />
-                    </td>
-                    <td style={{ width: 50, paddingRight: 5, textAlign: 'center' }}>
-                      <TooltipElement
-                        content={`${formatPercentage(ability.wasted / totalWasted)} %`}
-                      >
-                        {ability.wasted}
-                      </TooltipElement>
-                    </td>
-                    <td style={{ width: '30%' }}>
-                      <div
-                        className="performance-bar"
-                        style={{ width: `${(ability.wasted / totalWasted) * 100}%` }}
-                      />
-                    </td>
-                  </tr>
-                ))}
+            {generated?.map((ability) => (
+              <tr key={ability.abilityId}>
+                <td style={{ width: '30%' }}>
+                  {ability.abilityId === SPELLS.RUNE_1.id && (
+                    <>
+                      <SpellIcon id={ability.abilityId} noLink />
+                      <Trans id="deathknight.shared.runeBreakdown.passive">
+                        Passive Rune regeneration
+                      </Trans>
+                    </>
+                  )}
+                  {ability.abilityId !== SPELLS.RUNE_1.id && <SpellLink id={ability.abilityId} />}
+                </td>
+                <td style={{ width: 50, paddingRight: 5, textAlign: 'center' }}>
+                  <TooltipElement
+                    content={`${formatPercentage(ability.generated / totalGenerated)} %`}
+                  >
+                    {ability.generated}
+                  </TooltipElement>
+                </td>
+                <td style={{ width: '40%' }}>
+                  <div
+                    className="performance-bar"
+                    style={{ width: `${(ability.generated / totalGenerated) * 100}%` }}
+                  />
+                </td>
+                <td style={{ width: 50, paddingRight: 5, textAlign: 'center' }}>
+                  <TooltipElement content={`${formatPercentage(ability.wasted / totalWasted)} %`}>
+                    {ability.wasted}
+                  </TooltipElement>
+                </td>
+                <td style={{ width: '30%' }}>
+                  <div
+                    className="performance-bar"
+                    style={{ width: `${(ability.wasted / totalWasted) * 100}%` }}
+                  />
+                </td>
+              </tr>
+            ))}
           </tbody>
         </table>
         {showSpenders && (
           <table className="data-table">
             <thead>
               <tr>
-                <th>Ability</th>
-                <th colSpan="2">{resourceName} spent</th>
-                <th colSpan="2">Casts</th>
+                <th>
+                  <Trans id="deathknight.shared.runeBreakdown.ability">Ability</Trans>
```

Comment:
common labels like this should really be a part of a common set of translation strings. we've been pretty lax about that in the past because some translations are better than none, but having to translate `Ability` for every module that uses the label is not ideal

Reply:
👍🏼 That thought occurred to me too, I'll go through it and extract some things that could be common across specs.